code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +467,551 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+	)
+
+/datum/action/ability/activable/xeno/tail_lash/use_ability(atom/target)
+	xeno_owner.face_atom(target)
+
+	var/turf/lower_left
+	var/turf/upper_right
+	switch(xeno_owner.dir)
+		if(NORTH)
+			lower_left = locate(xeno_owner.x - 1, xeno_owner.y + 1, xeno_owner.z)
+			upper_right = locate(xeno_owner.x + 1, xeno_owner.y + 2, xeno_owner.z)
+		if(SOUTH)
+			lower_left = locate(xeno_owner.x - 1, xeno_owner.y - 2, xeno_owner.z)
+			upper_right = locate(xeno_owner.x + 1, xeno_owner.y - 1, xeno_owner.z)
+		if(WEST)
+			lower_left = locate(xeno_owner.x - 2, xeno_owner.y - 1, xeno_owner.z)
+			upper_right = locate(xeno_owner.x - 1, xeno_owner.y + 1, xeno_owner.z)
+		if(EAST)
+			lower_left = locate(xeno_owner.x + 1, xeno_owner.y - 1, xeno_owner.z)
+			upper_right = locate(xeno_owner.x + 2, xeno_owner.y + 1, xeno_owner.z)
+
+	for(var/turf/affected_tile in block(lower_left, upper_right))
+		affected_tile.Shake(duration = 0.1 SECONDS)
+		for(var/atom/movable/affected AS in affected_tile)
+			if(!ishuman(affected))
+				continue
+			if(affected.move_resist >= MOVE_FORCE_OVERPOWERING)
+				continue
+			var/mob/living/carbon/human/affected_human = affected
+			if(affected_human.stat == DEAD)
+				continue
+			affected_human.Paralyze(0.1 SECONDS)
+			affected_human.apply_effect(0.1 SECONDS, WEAKEN)
+			var/throwlocation = affected_human.loc
+			for(var/x in 1 to 2)
+				throwlocation = get_step(throwlocation, owner.dir)
+			affected_human.throw_at(throwlocation, 2, 1, owner, TRUE)
+			// Preventing this from infinitely being used as a cost-free throw against the crit.
+			if(affected_human.stat == UNCONSCIOUS)
+				RegisterSignal(affected_human, COMSIG_MOVABLE_MOVED, PROC_REF(on_movement_while_thrown))
+				RegisterSignal(affected_human, COMSIG_MOVABLE_POST_THROW, PROC_REF(on_throw_end))
+
+	xeno_owner.spin(4, 1)
+	xeno_owner.emote("tail")
+	playsound(xeno_owner, 'sound/weapons/alien_claw_block.ogg', 50, 1)
+	succeed_activate()
+	add_cooldown()
+
+/// Does brute/oxygen damage to unconscious humans if they move.
+/datum/action/ability/activable/xeno/tail_lash/proc/on_movement_while_thrown(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/human_mob = source
+	if(human_mob.stat != UNCONSCIOUS)
+		UnregisterSignal(human_mob, COMSIG_MOVABLE_MOVED)
+		UnregisterSignal(human_mob, COMSIG_MOVABLE_POST_THROW)
+		return
+	if(human_mob.adjustOxyLoss(HUMAN_CRITDRAG_OXYLOSS))
+		return
+	INVOKE_ASYNC(human_mob, TYPE_PROC_REF(/mob/living/carbon/human, adjustBruteLoss), HUMAN_CRITDRAG_OXYLOSS)
+
+/// Removes signals for COMSIG_MOVABLE_MOVED and COMSIG_MOVABLE_POST_THROW for thrown humans.
+/datum/action/ability/activable/xeno/tail_lash/proc/on_throw_end(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(source, COMSIG_MOVABLE_POST_THROW)
+
+// ***************************************
+// *********** Advance
+// ***************************************
+/datum/action/ability/activable/xeno/advance
```

Comment:
As above, just give this a new name, its a typepath we don't care if its called prae_super_cool_advance, dont just change other stuff because.